pkg/controller/autoimport/autoimport_controller.go

```diff
@@ -78,100 +77,233 @@ func (r *ReconcileAutoImport) Reconcile(ctx context.Context, request reconcile.R
+				fmt.Sprintf("Expect 2 manifestwork, but got %v", len(manifestWorks)),
+			), rc, nil
 	}
 
-	importClient, restMapper, importErr := helpers.GenerateClientFromSecret(autoImportSecret)
-	switch {
-	case importErr != nil:
-		// failed to generate import client with auto-import sercet, will reduce the auto-import secret retry times and reconcile again
-	case importErr == nil:
-		if val, ok := autoImportSecret.Labels[restoreLabel]; ok && strings.EqualFold(val, "true") {
-			// only update the bootstrap secret on the managed cluster with the auto-import-secret
-			importErr = helpers.UpdateManagedClusterBootstrapSecret(importClient, importSecret, r.recorder)
-		} else {
-			importErr = helpers.ImportManagedClusterFromSecret(importClient, restMapper, r.recorder, importSecret)
-		}
+	rc.current++
+	r.recorder.Eventf("RetryToImportCluster", "Retry to import cluster %s, %d/%d",
+		autoImportSecret.Namespace, rc.current, rc.total)
+
+	importClient, restMapper, err := helpers.GenerateClientFromSecret(autoImportSecret)
+	if err != nil {
+		return reconcile.Result{},
+			newManagedClusterImportSucceededCondition(
+				metav1.ConditionFalse,
+				constants.ConditionReasonAutoImportSecretInvalid,
+				fmt.Sprintf("Retry to import managed cluster, retry times: %d/%d, generate kube client by secret error: %v",
+					rc.current, rc.total, err),
+			), rc, err
 	}
 
-	if importErr != nil {
-		if err := helpers.UpdateManagedClusterStatus(
-			r.client,
-			r.recorder,
-			managedClusterName,
-			metav1.Condition{
-				Type:   "ManagedClusterImportSucceeded",
-				Status: metav1.ConditionFalse,
-				Message: fmt.Sprintf("Unable to import managed cluster %s with auto-import-secret: %s",
-					managedClusterName, importErr.Error()),
-				Reason: "FailedToDeployKlusterlet",
-			},
-		); err != nil {
-			return reconcile.Result{}, err
+	if val, ok := autoImportSecret.Labels[restoreLabel]; ok && strings.EqualFold(val, "true") {
+		// only update the bootstrap secret on the managed cluster with the auto-import-secret
+		err = helpers.UpdateManagedClusterBootstrapSecret(importClient, importSecret, r.recorder)
+	} else {
+		err = helpers.ImportManagedClusterFromSecret(importClient, restMapper, r.recorder, importSecret)
+	}
+	if err != nil {
+		if containAuthError(err) {
+			// return an invalid auto import secret reason so that the user knows that
+			// a correct secret needs to be re-provided
+			return reconcile.Result{RequeueAfter: 10 * time.Second},
+				newManagedClusterImportSucceededCondition(
+					metav1.ConditionFalse,
+					constants.ConditionReasonAutoImportSecretInvalid,
+					fmt.Sprintf("Retry to import managed cluster, retry times: %d/%d, error: %v",
+						rc.current, rc.total, err),
+				), rc, err
 		}
 
-		// failed to apply the import secrect, reduce the retry times and reconcile again
-		retryErr := helpers.UpdateAutoImportRetryTimes(ctx, r.kubeClient, r.recorder, autoImportSecret.DeepCopy())
```

Comment:
how about to enhance the `UpdateAutoImportRetryTimes`? because this is also is used by hosted import controller, I think we should need a  consistent way to handle the auto import retry.

For `UpdateAutoImportRetryTimes`, we may consider to add a currentRetryTimes in the autoimport secret data instead of reducing the `autoImportRetry` directly